org/postgresql/jdbc2/AbstractJdbc2Statement.java

```diff
@@ -1866,15 +1847,191 @@ else if (in instanceof InputStream)
             case Types.DISTINCT:
                 bindString(parameterIndex, in.toString(), Oid.UNSPECIFIED);
                 break;
-	        case Types.OTHER:
-	            if (in instanceof PGobject)
-	                setPGobject(parameterIndex, (PGobject)in);
-	            else
+            case Types.OTHER:
+                if (in instanceof PGobject)
+                    setPGobject(parameterIndex, (PGobject)in);
+                else
                     bindString(parameterIndex, in.toString(), Oid.UNSPECIFIED);
-	            break;
-	        default:
-	            throw new PSQLException(GT.tr("Unsupported Types value: {0}", targetSqlType), PSQLState.INVALID_PARAMETER_TYPE);
+                break;
+            default:
+                throw new PSQLException(GT.tr("Unsupported Types value: {0}", targetSqlType), PSQLState.INVALID_PARAMETER_TYPE);
+        }
+    }
+
+    private static String asString(final Clob in) throws SQLException {
+        return in.getSubString(1, (int) in.length());
+    }
+
+    private static int castToInt(final Object in) throws SQLException {
+        try {
+            if (in instanceof String)
+                return Integer.parseInt((String) in);
+            if (in instanceof Number)
+                return ((Number) in).intValue();
+            if (in instanceof java.util.Date)
+                return (int) ((java.util.Date) in).getTime();
+            if (in instanceof Boolean)
+                return (Boolean) in ? 1 : 0;
+            if (in instanceof Clob)
+                return Integer.parseInt(asString((Clob) in));
+            if (in instanceof Character)
+                return Integer.parseInt(in.toString());
+        } catch (final Exception e) {
+            throw cannotCastException(in.getClass().getName(), "int", e);
+        }
+        throw cannotCastException(in.getClass().getName(), "int");
+    }
+
+    private static short castToShort(final Object in) throws SQLException {
```

Comment:
@davecramer , You are wrong here.
Java has no multiple dispatch.
In other words, since _compile-time_ variable type is `Object`, runtime would not be able to pick the "correct" overload.
So, either `instanceof` or some hashmap is required.